src/functionalTest/groovy/com/autonomousapps/jvm/BundleSpec.groovy

```diff
@@ -71,7 +72,22 @@ final class BundleSpec extends AbstractJvmSpec {
 
     // TODO reason needs to be updated to show that the -jvm variant is used
     expect:
-    build(gradleVersion, gradleProject.rootDir, ':consumer:reason', '--id', 'com.squareup.okio:okio')
+    build(gradleVersion, gradleProject.rootDir, ':consumer:reason', '--id', 'com.squareup.okio:okio-jvm')
+
+    where:
+    gradleVersion << [GradleVersion.current()]
+  }
+
+  def "reason fails when there is dependency filtering ambiguity"() {
+    given:
+    def project = new BundleKmpProject2()
+    gradleProject = project.gradleProject
+
+    when:
+    def result = buildAndFail(gradleVersion, gradleProject.rootDir, ':consumer:reason', '--id', 'com.squareup.okio:okio')
+
+    then:
+    assertThat(result.output).contains("> Coordinates 'com.squareup.okio:okio' matches more than 1 dependency [com.squareup.okio:okio-jvm:3.0.0, com.squareup.okio:okio:3.0.0]")
 
```

Comment:
We don't want this. We want to be able to reference a dependency by its identifier alone. If the user passes in `com.squareup.okio:okio`, it should unambiguously match `com.squareup.okio:okio:3.0.0`.

Reply:
You are totally right. I've implemented it this way for project modules, but forgot about non-project 🤦‍♂️.
BTW thanks for integration testing - it highlighted the problem (but I still missed it 🤦‍♂️).
Will address all comments in a couple of days - it seems I've caught flu and need to restore.

Reply:
done